core/storage/base_model/gae_models.py

```diff
@@ -891,15 +909,27 @@ def get_snapshot_id(cls, instance_id: str, version_number: int) -> str:
         return '%s%s%s' % (
             instance_id, VERSION_DELIMITER, version_number)
 
+    def _prepare_additional_models(self) -> Mapping[str, BaseModel]:
+        """Prepares additional models needed for the commit process. The dict
+        is empty by default  and this method should be overriden in models that
+        inherit from VersionedModel.
+
+        Returns:
+            dict(str, BaseModel). Additional models needed for
+            the commit process.
+        """
+        return {}
+
     # TODO(#13523): Change 'commit_cmds' to domain object/TypedDict to
     # remove Any from type-annotation below.
-    def _trusted_commit(
-            self,
-            committer_id: str,
-            commit_type: str,
-            commit_message: str,
-            commit_cmds: List[Dict[str, Any]]
-    ) -> None:
+    def compute_models_to_commit(
+        self,
+        committer_id: str,
+        commit_type: str,
+        commit_message: str,
+        commit_cmds: List[Dict[str, Any]],
+        unused_additional_models: Mapping[str, BaseModel]
```

Comment:
What is Mapping -- is it the same as Dict? If so why don't we just call it Dict?

(If it's supposed to be Dict, why didn't the linter catch this?)

Reply:
Mapping is a bit different, it allows the values to be child of the thing that is in the second place, so for `Mapping[str, BaseModel]` values can be models that inherit from `BaseModel`. If it was `Dict[str, BaseModel]` this wouldn't be allowed.

https://stackoverflow.com/questions/52487663/python-type-hints-typing-mapping-vs-typing-dict#comment113661999_52487800

Reply:
Added a comment about this.